File: auto_clicker.py
import time
import threading
import pynput.mouse as mouse
import pynput.keyboard as keyboard

class AutoClicker:
    def __init__(self):
        self.clicking = False
        self.running = True
        self.click_delay = 0.05  # 클릭 간격 (초)
        self.mouse_controller = mouse.Controller()
        self.mouse_listener = None
        self.keyboard_listener = None
        
    def on_click(self, x, y, button, pressed):
        if button == mouse.Button.left:
            if pressed:
                self.clicking = True
                threading.Thread(target=self.auto_click, daemon=True).start()
            else:
                self.clicking = False
                
    def auto_click(self):
        while self.clicking and self.running:
            self.mouse_controller.click(mouse.Button.left)
            time.sleep(self.click_delay)
            
    def on_key(self, key):
        try:
            if key == keyboard.Key.f1:  # F1으로 속도 증가
                self.click_delay = max(0.01, self.click_delay - 0.01)
                print(f"클릭 속도 증가: {1/self.click_delay:.1f} CPS")
            elif key == keyboard.Key.f2:  # F2로 속도 감소
                self.click_delay = min(0.5, self.click_delay + 0.01)
                print(f"클릭 속도 감소: {1/self.click_delay:.1f} CPS")
            elif key == keyboard.Key.esc:  # ESC로 프로그램 종료
                print("프로그램 종료...")
                self.stop()
                return False
        except AttributeError:
            pass
            
    def start(self):
        print("오토클리커 시작!")
        print("사용법:")
        print("- 마우스 좌클릭을 누르고 있으면 자동 연타")
        print("- F1: 클릭 속도 증가")
        print("- F2: 클릭 속도 감소")
        print("- ESC: 프로그램 종료")
        print(f"현재 클릭 속도: {1/self.click_delay:.1f} CPS")
        
        self.mouse_listener = mouse.Listener(on_click=self.on_click)
        self.keyboard_listener = keyboard.Listener(on_press=self.on_key)
        
        self.mouse_listener.start()
        self.keyboard_listener.start()
        
        self.keyboard_listener.join()
        
    def stop(self):
        self.running = False
        self.clicking = False
        if self.mouse_listener:
            self.mouse_listener.stop()

if __name__ == "__main__":
    clicker = AutoClicker()
    clicker.start()

File: needaimbot/mouse/AutoClicker.cpp
#include "AutoClicker.h"
#include <iostream>

AutoClicker::AutoClicker() 
    : isRunning(false), isClicking(false), clickDelay(50) {
}

AutoClicker::~AutoClicker() {
    Stop();
}

void AutoClicker::ClickLoop() {
    while (isRunning) {
        if (isClicking) {
            INPUT input = {0};
            input.type = INPUT_MOUSE;
            input.mi.dwFlags = MOUSEEVENTF_LEFTDOWN;
            SendInput(1, &input, sizeof(INPUT));
            
            Sleep(15);  // Reduce CPU usage - 15ms is still very responsive
            
            input.mi.dwFlags = MOUSEEVENTF_LEFTUP;
            SendInput(1, &input, sizeof(INPUT));
            
            Sleep(clickDelay - 5);
        } else {
            Sleep(10);
        }
    }
}

void AutoClicker::InputLoop() {
    bool leftButtonWasPressed = false;
    
    while (isRunning) {
        bool leftButtonPressed = (GetAsyncKeyState(VK_LBUTTON) & 0x8000) != 0;
        
        if (leftButtonPressed && !leftButtonWasPressed) {
            isClicking = true;
        } else if (!leftButtonPressed && leftButtonWasPressed) {
            isClicking = false;
        }
        
        leftButtonWasPressed = leftButtonPressed;
        
        // F1 키로 속도 증가
        if (GetAsyncKeyState(VK_F1) & 0x0001) {
            int newDelay = std::max(10, clickDelay.load() - 10);
            SetClickDelay(newDelay);
            std::cout << "Click speed increased: " << (1000.0 / newDelay) << " CPS\n";
        }
        
        // F2 키로 속도 감소
        if (GetAsyncKeyState(VK_F2) & 0x0001) {
            int newDelay = std::min(500, clickDelay.load() + 10);
            SetClickDelay(newDelay);
            std::cout << "Click speed decreased: " << (1000.0 / newDelay) << " CPS\n";
        }
        
        // ESC 키로 종료
        if (GetAsyncKeyState(VK_ESCAPE) & 0x0001) {
            std::cout << "Stopping auto clicker...\n";
            Stop();
            break;
        }
        
        Sleep(15);  // Reduce CPU usage - 15ms is still very responsive
    }
}

void AutoClicker::Start() {
    if (isRunning) return;
    
    isRunning = true;
    
    std::cout << "Auto Clicker Started!\n";
    std::cout << "Usage:\n";
    std::cout << "- Hold left mouse button for auto clicking\n";
    std::cout << "- F1: Increase click speed\n";
    std::cout << "- F2: Decrease click speed\n";
    std::cout << "- ESC: Stop program\n";
    std::cout << "Current speed: " << (1000.0 / clickDelay) << " CPS\n";
    
    clickThread = std::thread(&AutoClicker::ClickLoop, this);
    inputThread = std::thread(&AutoClicker::InputLoop, this);
}

void AutoClicker::Stop() {
    isRunning = false;
    isClicking = false;
    
    if (clickThread.joinable()) {
        clickThread.join();
    }
    if (inputThread.joinable()) {
        inputThread.join();
    }
}

void AutoClicker::SetClickDelay(int delayMs) {
    clickDelay = delayMs;
}